### PR TITLE
[NPU] Update tools

### PR DESCRIPTION
**Pull request by XinWangIntel**
### Details:
Fix coverity issue for compile_tool.
Support template plugin for single-image-test.

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Review by PatrikStepan — APPROVED** [review]

**Review by Maxim-Doronin — APPROVED** [review]
LGTM. Please review npu_trigger in ci/jenkins pipeline carefully before merging

**Comment by PatrikStepan** [comment]
There is a known issue with npu tests on linux. They all pass on windows. Adding this to the merge queue to unblock other PRs.

**Review by razvanapetroaie — APPROVED** [review]

**Pull request merged**
